samlsp/middleware_test.go

```diff
@@ -76,8 +76,11 @@ func (test *MiddlewareTest) SetUpTest(c *C) {
 			IDPMetadata: &saml.EntityDescriptor{},
 			Logger:      logger.DefaultLogger,
 		},
-		CookieName:   "ttt",
-		CookieMaxAge: time.Hour * 2,
+		CookieName:       "ttt",
+		CookieMaxAge:     time.Hour * 2,
+		JwtSigningMethod: jwt.SigningMethodHS256,
+		JwtSigningKey:    x509.MarshalPKCS1PrivateKey(test.Key),
```

Comment:
I'm a little confused by this ... crypto.PrivateKey is an interface{} but we're assigning a `[]byte` here. Is that an error?

Reply:
It is indeed confusing, but this is the way private key works, `crypto.PrivateKey` is a generic `interface{}` for any type of key.
Here we are using `jwt` with either `rsa` (`SigningMethodRSA`) or `hmac` (`SigningMethodHMAC`) implementing `jwt.SigningMethod`

To comply with `jwt.SigningMethod` interface, keys must be interfaces ([signing_method.go](https://github.com/dgrijalva/jwt-go/blob/a539ee1a749a2b895533f979515ac7e6e0f5b650/signing_method.go#L13))

Then signing methods for both RSA and HMAC checks that the key is the expected type:
- [rsa.go](https://github.com/dgrijalva/jwt-go/blob/a539ee1a749a2b895533f979515ac7e6e0f5b650/rsa.go#L82) checks that key is a `*rsa.PrivateKey`
- [hmac.go](https://github.com/dgrijalva/jwt-go/blob/a539ee1a749a2b895533f979515ac7e6e0f5b650/hmac.go#L82) checks that key is a `[]byte`